consensus/XDPoS/engines/engine_v2/forensics.go

```diff
@@ -17,36 +21,68 @@ type ForensicProof struct {
 	QcWithLargerRoundAddresses  []common.Address
 }
 
+// Forensics instance. Placeholder for future properties to be added
 type Forensics struct {
-	ReceiverCh <-chan utils.QuorumCert
-	Abort      chan<- struct{}
+	HighestCommittedQCs []utils.QuorumCert
 }
 
 // Initiate a forensics process
-func (x *XDPoS_v2) AttachForensics() {
-	receiver := make(chan utils.QuorumCert)
-	abort := make(chan struct{})
-
-	go func() {
-		for {
-			// A real event arrived, process interesting content
-			select {
-			case quorumCert := <-receiver:
-				x.ProcessForensics(quorumCert)
-			case <-abort:
-				return
+func NewForensics() *Forensics {
+	return &Forensics{}
+}
+
+/*
+	Entry point for processing forensics.
+	Triggered once processQC is successfully.
+	Forensics runs in a seperate go routine as its no system critical
+	Link to the flow diagram: https://hashlabs.atlassian.net/wiki/spaces/HASHLABS/pages/97878029/Forensics+Diagram+flow
+*/
+func (f *Forensics) ProcessForensics(chain consensus.ChainReader, incomingQC utils.QuorumCert) {
+	log.Info("Received a QC in forensics", "QC", incomingQC)
+}
+
+// Set the forensics committed QCs list. The order is from grandparent to current header. i.e it shall follow the QC in its header as follow [hcqc1, hcqc2, hcqc3]
+func (f *Forensics) SetCommittedQCs(headers []types.Header, incomingQC utils.QuorumCert) error {
+	// highestCommitQCs is an array, assign the parentBlockQc and its child as well as its grandchild QC into this array for forensics purposes.
+	if len(headers) != 2 {
+		log.Error("[SetCommittedQcs] Received input length not equal to 2", len(headers))
+		return fmt.Errorf("Received headers length not equal to 2 ")
+	}
+
+	var committedQCs []utils.QuorumCert
+	for i, h := range headers {
```

Comment:
you can use below loop to remove `i != 0` 
```
for i := 1; i < 2; i++ {
```

Reply:
the i != 0 is just one condition. I still need to loop through from index 0